### PR TITLE
Improve/performance

### PR DESCRIPTION
**Pull request by RamonGilabert**
First the memory performance improvement:

Before:

screen shot 2015-08-31 at 2 25 37 pm

After:

screen shot 2015-08-31 at 2 22 55 pm

Big changes:

- ASCollectionView -> ASTableView done that to check if it's easier in the project to manipulate cells, might go back to ASCollectionView, I've seen that it has more work behind in AsyncDisplayKit.
- Drop Haneke, we don't need it anymore.
- Use ASNetworkImageNode.
- Use a bunch of layers to be rendered instead of render the node as views or something else.

### CODE REVIEW

**Comment by richardoti** [comment]
Wow! Dropped more than twice in size!

File: Source/Nodes/Defaults/AttachmentGridNode.swift [hunk 1]
```diff
@@ -42,23 +42,17 @@ public class AttachmentGridNode: PostComponentNode {
     var lastImageSize: CGSize?
 
     for (index, attachment) in enumerate(attachments) {
-      let imageNode = ASImageNode()
+      let imageNode = ASNetworkImageNode()
```
Comment (richardoti) [inline Source/Nodes/Defaults/AttachmentGridNode.swift]: Could we use it with collection view?

Comment (RamonGilabert) [inline Source/Nodes/Defaults/AttachmentGridNode.swift]: Yes

**Comment by richardoti** [comment]
Overall, looks good to me, I doubt that this refactoring may lead to any bugs as it is just rewriting to TableView. I like the memory footprint decrease :+1: 
I would like @vadymmarkov and @zenangst to comment on this, as they might have some suggestions.

File: Source/Nodes/Defaults/AttachmentGridNode.swift [hunk 2]
```diff
@@ -113,7 +107,7 @@ public class AttachmentGridNode: PostComponentNode {
 
   func sizeForThumbnailAtIndex(index: Int) -> CGSize {
     var size = CGSize(width: width, height: height)
-
+
```
Comment (zenangst) [inline Source/Nodes/Defaults/AttachmentGridNode.swift]: Such space

**Comment by RamonGilabert** [comment]
Again, the change to ASTableView it's not definitive, want to test it more with more stuff into the Wall and see the change in performance.

**Comment by richardoti** [comment]
Got crash after clean build:
```
dyld: Library not loaded: @rpath/Haneke.framework/Haneke
  Referenced from: /Users/[USERNAME]/Library/Developer/CoreSimulator/Devices/A65EEBB2-A5D2-4E00-A642-70786A16CE1F/data/Containers/Bundle/Application/E891F738-37A4-48E5-8FE6-DA13BE6EE67D/InfiniteScroll.app/Frameworks/Wall.framework/Wall
  Reason: image not found
(lldb) 
```

**Comment by RamonGilabert** [comment]
:smile: Haneke shouldn't be there

**Comment by vadymmarkov** [comment]
@RamonGilabert Looks awesome!

**Comment by richardoti** [comment]
```Swift
  public struct ImageCache {
    public static var format = "wall-thumbnails"
    public static var storage: Haneke.Cache<UIImage> {
      let cache = Shared.imageCache
      let format = Format<UIImage>(name: self.format, diskCapacity: 20 * 1024 * 1024)
      cache.addFormat(format)

      return cache
    }
  }
```

**Comment by zenangst** [comment]
whoops!

**Pull request merged**